### PR TITLE
fix for giant fonts in RTS

### PR DESCRIPTION
**Pull request by adipose**

### CODE REVIEW

File: src/Subtitles/RTS.cpp [hunk 1]
```diff
@@ -653,9 +677,20 @@ bool CText::CreatePath()
             }
         }
     }
-
     SelectFont(g_hDC, hOldFont);
 
+    if (m_style.fontSize > MAXGDIFONTSIZE) {
+        double fact = m_style.fontSize / MAXGDIFONTSIZE;
+        for (ptrdiff_t i = 0; i < mPathPoints; i++) {
+            double x, y, z, xx, yy, zz;
```
Comment (clsid2) [inline src/Subtitles/RTS.cpp]:
```
mpPathPoints[i].x *= fact;
mpPathPoints[i].y *= fact;
```

Comment (adipose) [reply src/Subtitles/RTS.cpp]: Yeah that's better. I copied some other code and didn't simplify it.

Should put this in a function to keep it clean probably.

**Comment by clsid2** [comment]
Wouldn't it be simpler to do this once when adding a style:
```
if (style.fontSize > 39999) {
    int fact = style.fontSize / 39999;
    style.fontSize = 39999;
    style.fontScaleX *= fact;
    style.fontScaleY *= fact;
}
```

**Comment by adipose** [comment]
Probably.  I wasn't sure I could depend on fontscale to do it properly, but I imagine it might work.

**Comment by adipose** [comment]
> Wouldn't it be simpler to do this once when adding a style:
> 
> ```
> if (style.fontSize > 39999) {
>     int fact = style.fontSize / 39999;
>     style.fontSize = 39999;
>     style.fontScaleX *= fact;
>     style.fontScaleY *= fact;
> }
> ```

That did not work.  It scales ok but it doesn't scale around the right origin.

**Pull request merged**